contrib/log/slog/slog.go

```diff
@@ -1,40 +1,73 @@
 package slog
 
 import (
+	"context"
 	"fmt"
 	"log/slog"
+	"runtime"
+	"time"
 
 	"github.com/go-kratos/kratos/v2/log"
 )
 
 var _ log.Logger = (*Logger)(nil)
 
 type Logger struct {
-	log *slog.Logger
+	handler slog.Handler
 }
 
-func NewLogger(slog *slog.Logger) log.Logger {
-	return &Logger{slog}
+func NewLogger(handler slog.Handler) log.Logger {
+	return &Logger{handler}
+}
+
+func (l *Logger) log(ctx context.Context, level slog.Level, msg string, args ...any) {
+	if l == nil || !l.handler.Enabled(ctx, level) {
+		return
+	}
+	var pc uintptr
+	var pcs [1]uintptr
+	// skip [runtime.Callers, this function, this function's caller]
+	runtime.Callers(3, pcs[:])
+	pc = pcs[0]
+	r := slog.NewRecord(time.Now(), level, msg, pc)
+	r.Add(args...)
+	_ = l.handler.Handle(ctx, r)
+}
+
+func (l *Logger) Debug(msg string, args ...any) {
+	l.log(context.Background(), slog.LevelDebug, msg, args...)
+}
+
+func (l *Logger) Warn(msg string, args ...any) {
+	l.log(context.Background(), slog.LevelWarn, msg, args...)
+}
+
+func (l *Logger) Info(msg string, args ...any) {
+	l.log(context.Background(), slog.LevelInfo, msg, args...)
+}
+
+func (l *Logger) Error(msg string, args ...any) {
+	l.log(context.Background(), slog.LevelError, msg, args...)
 }
 
 func (l *Logger) Log(level log.Level, keyvals ...interface{}) error {
 	keylen := len(keyvals)
 	if keylen == 0 || keylen%2 != 0 {
-		l.log.Warn(fmt.Sprint("Keyvalues must appear in pairs: ", keyvals))
+		l.Warn(fmt.Sprint("Keyvalues must appear in pairs: ", keyvals))
 		return nil
 	}
 
 	switch level {
 	case log.LevelDebug:
-		l.log.Debug("", keyvals...)
+		l.Debug("", keyvals...)
```

Comment:
Consider to add a special case for DefaultMessageKey? So users can use `helper.Info("info")` to print "msg=info"(not msg='' msg=info).